.github/workflows/docker.yml

```diff
@@ -0,0 +1,73 @@
+# NOTE: This name appears in GitHub's Checks API and in workflow's status badge.
+name: docker
+
+# Trigger the workflow when:
+on:
+  # A push occurs to one of the matched branches.
+  push:
+    branches:
+      - master
+
+jobs:
+
+  build-benchmarking-image:
+    # NOTE: This name appears in GitHub's Checks API.
+    name: build-benchmarking-images
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          # Check out pull request's HEAD commit instead of the merge commit.
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Determine tag name
+        id: determine_tag
+        shell: bash
+        run: |
+          if [[ -z $GITHUB_BASE_REF ]]; then
+            # On master/stable branches.
+            branch=${GITHUB_REF#refs/heads/}
+          else
+            # On pull request branches.
+            branch=pr-$(git describe --always --match '' --abbrev=7)
+          fi
+          branch=${branch//\//-}
+          echo "::set-output name=tag::$(echo ${branch})"
+          echo "::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')"
+
+      # Build benchmark artifacts.
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+      - name: Set up Go 1.16
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: "1.16.x"
+      - name: Install Oasis dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install bubblewrap build-essential libseccomp-dev libssl-dev protobuf-compiler
+      - name: Build benchmarks
+        working-directory: tests/benchmark
+        run: ./build-benchmarks.sh
+
+      # Build and push benchmarks docker image.
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_ACCESS_TOKEN }}
+      - name: "Rebuild oasisprotocol/oasis-sdk-benchmark:${{ steps.determine_tag.outputs.tag }}"
+        uses: docker/build-push-action@v2.5.0
+        with:
+          context: .
+          file: tests/benchmark/Dockerfile
```

Comment:
Could we make it so that the Oasis Core version comes from the const file that is also used for tests? To avoid having the version hardcoded in too many places.

Reply:
good idea, fixed